libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -876,17 +895,54 @@ void AC_Fence::record_breach(uint8_t fence_type)
     _breached_fences |= fence_type;
 }
 
-/// clear_breach - update breach bitmask, time and count
+/// record_margin_breach - update margin_breach bitmask, time and count
+void AC_Fence::record_margin_breach(uint8_t fence_type)
+{
+    // if we haven't already breached a margin limit, update the breach time
+    if (!(_breached_fence_margins & fence_type)) {
+        const uint32_t now = AP_HAL::millis();
+
+        // emit a message indicated we're newly-breached, but not too often
+        if (option_enabled(OPTIONS::MARGIN_BREACH)
+            && AP_HAL::timeout_expired(_last_margin_breach_notify_sent_ms, now, 1000U)) {
+            _last_margin_breach_notify_sent_ms = now;
+            print_fence_message("outside margin", _breached_fence_margins | fence_type);
```

Comment:
i'm not sure "outside margin" and "inside margin" are clear, when I first read it I wondered if "outside margin" is "almost breached" or "all OK"
@rmackay9 ?

Reply:
yes, we should try and be clear and we support both include and exclude fences.  Maybe we should used "breach margin" and "cleared margin"?

Reply:
> yes, we should try and be clear and we support both include and exclude fences. Maybe we should used "breach margin" and "cleared margin"?

Perhaps the other way around - "margin breach", "margin clear".

Reply:
Ok I have changed to "xxx fence close" and "xxx fence cleared margin breach"